lib/sinon/match.js

```diff
@@ -231,37 +231,57 @@ match.array.deepEquals = function (expectation) {
     return match(function (actual) {
         // Comparing lengths is the fastest way to spot a difference before iterating through every item
         var sameLength = actual.length === expectation.length;
-        return typeOf(actual) === "array" && sameLength && every.call(actual, function (element, index) {
+        return typeOf(actual) === "array" && sameLength && every(actual, function (element, index) {
             return expectation[index] === element;
         });
-    }, "deepEquals([" + arrayToString.call(expectation) + "])");
+    }, "deepEquals([" + iterableToString(expectation) + "])");
 };
 
 match.array.startsWith = function (expectation) {
     return match(function (actual) {
-        return typeOf(actual) === "array" && every.call(expectation, function (expectedElement, index) {
+        return typeOf(actual) === "array" && every(expectation, function (expectedElement, index) {
             return actual[index] === expectedElement;
         });
-    }, "startsWith([" + arrayToString.call(expectation) + "])");
+    }, "startsWith([" + iterableToString(expectation) + "])");
 };
 
 match.array.endsWith = function (expectation) {
     return match(function (actual) {
         // This indicates the index in which we should start matching
         var offset = actual.length - expectation.length;
 
-        return typeOf(actual) === "array" && every.call(expectation, function (expectedElement, index) {
+        return typeOf(actual) === "array" && every(expectation, function (expectedElement, index) {
             return actual[offset + index] === expectedElement;
         });
-    }, "endsWith([" + arrayToString.call(expectation) + "])");
+    }, "endsWith([" + iterableToString(expectation) + "])");
 };
 
 match.array.contains = function (expectation) {
     return match(function (actual) {
-        return typeOf(actual) === "array" && every.call(expectation, function (expectedElement) {
+        return typeOf(actual) === "array" && every(expectation, function (expectedElement) {
             return indexOf.call(actual, expectedElement) !== -1;
         });
-    }, "contains([" + arrayToString.call(expectation) + "])");
+    }, "contains([" + iterableToString(expectation) + "])");
+};
+
+match.map = match.typeOf("map");
+
+match.map.deepEquals = function (expectation) {
```

Comment:
Name anonymous function to aid debugging (ref [previous review](https://github.com/sinonjs/sinon/pull/1215#discussion_r94131858)).